### PR TITLE
feat(2245): implemented user analytics with matomo

### PR DESCRIPTION
**Pull request by akhilmhdh**
# Closes

1. 2245

# Explain the feature/fix

1. Implemented  user analytics in analytics service
2. Integrated with matomo
3. Changed sentry crashlytics api

## Does this PR introduce a breaking change

Yes

Changed crashlytics API

## Screenshots

<details>
<summary>View Screenshots</summary>

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [ ] Does the change have appropriate unit tests?
- [ ] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demoed and the design review approved?

### CODE REVIEW

**Review by riginoommen — CHANGES_REQUESTED** [review]

File: packages/analytics-service/src/datasource/matomoAPI.ts [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+  TUserActionsData,
+  TUserGeographyData,
+  TUserHistoryDTO,
+  TUserMetricsDTO,
+  TUserVisitorData,
+  TUserVisitorFrequencyData,
+  UpdateWebsiteDTO,
+} from './types';
+
+// http://matamo-rhn-support-akmohan-dev.apps.sandbox.x8i5.p1.openshiftapps.com/index.php?date=2022-08-29&module=API&format=json&method=SitesManager.addSite&segment=&idSite=1&period=day
```
[the remark this answers is not in the document]
Comment (akhilmhdh) [reply packages/analytics-service/src/datasource/matomoAPI.ts]: Removed. It was for a reference my bad

File: packages/analytics-service/src/app.ts [hunk 2]
```diff
@@ -37,10 +42,11 @@ export const setupMongoose = async () => {
 };
 
 export const graphqlSchema = mergeSchemas({
-  typeDefs: [analyticConfigSchema, crashlyticsSchema],
+  typeDefs: [analyticConfigSchema, crashlyticsSchema, userBehavoirSchema],
   resolvers: [
     analyticConfigResolver,
     crashlyticsResolver,
+    userBehavoirResolver,
```
[the remark this answers is not in the document]
Comment (akhilmhdh) [reply packages/analytics-service/src/app.ts]: Changed

File: packages/analytics-service/src/app.ts [hunk 3]
```diff
@@ -37,10 +42,11 @@ export const setupMongoose = async () => {
 };
 
 export const graphqlSchema = mergeSchemas({
-  typeDefs: [analyticConfigSchema, crashlyticsSchema],
+  typeDefs: [analyticConfigSchema, crashlyticsSchema, userBehavoirSchema],
```
[the remark this answers is not in the document]
Comment (akhilmhdh) [reply packages/analytics-service/src/app.ts]: Changed

File: packages/analytics-service/src/app.ts [hunk 4]
```diff
@@ -16,14 +16,19 @@ import { analyticConfigResolver } from './graph/analyticsConfig/resolver';
 import crashlyticsSchema from './graph/crashlytics/typedef.graphql';
 import { crashlyticsResolver } from './graph/crashlytics/resolver';
 
+import userBehavoirSchema from './graph/matomo/typedef.graphql';
+import { userBehavoirResolver } from './graph/matomo/resolver';
```
[the remark this answers is not in the document]
Comment (akhilmhdh) [reply packages/analytics-service/src/app.ts]: Changed

File: packages/analytics-service/src/app.ts [hunk 5]
```diff
@@ -16,14 +16,19 @@ import { analyticConfigResolver } from './graph/analyticsConfig/resolver';
 import crashlyticsSchema from './graph/crashlytics/typedef.graphql';
 import { crashlyticsResolver } from './graph/crashlytics/resolver';
 
+import userBehavoirSchema from './graph/matomo/typedef.graphql';
```
[the remark this answers is not in the document]
Comment (akhilmhdh) [reply packages/analytics-service/src/app.ts]: Changed

File: packages/analytics-service/.env.example [hunk 6]
```diff
@@ -2,6 +2,10 @@
 # NODE_ENV=
 # SENTRY_API_TOKEN=
 # SENTRY_API_URL=
+# SENTRY_API_ORG=
+# SENTRTY_API_TEAM=
```
Comment (riginoommen) [inline packages/analytics-service/.env.example:6]:
```suggestion
# SENTRY_API_TEAM=
```

File: packages/analytics-service/src/graph/errors.ts [hunk 7]
```diff
@@ -0,0 +1 @@
+export const ErrUnauthorizedAccess = new Error('Unauthorized access');
```
Comment (riginoommen) [inline packages/analytics-service/src/graph/errors.ts:1]: Use apollo server errors

https://www.apollographql.com/docs/apollo-server/data/errors/#forbidden

File: packages/analytics-service/.env.example (hunk 6 above)
Comment (riginoommen) [inline packages/analytics-service/.env.example:6]:
```suggestion
# SENTRY_API_TEAM=
```

**Review by riginoommen — CHANGES_REQUESTED** [review]

File: packages/analytics-service/.env.example (hunk 6 above)
Comment (akhilmhdh) [reply packages/analytics-service/.env.example]: Done

File: packages/analytics-service/src/graph/errors.ts (hunk 7 above)
Comment (akhilmhdh) [reply packages/analytics-service/src/graph/errors.ts]: Done

File: packages/analytics-service/.env.example (hunk 6 above)
Comment (akhilmhdh) [reply packages/analytics-service/.env.example]: Done

File: packages/analytics-service/src/graph/errors.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+import { GraphQLError } from 'graphql';
+
+export const ErrUnauthorizedAccess = new GraphQLError('Unautorized access', {
```
Comment (riginoommen) [inline packages/analytics-service/src/graph/errors.ts:3]:
```suggestion
export const ErrUnauthorizedAccess = new GraphQLError('Unauthorized access', {
```

**Review by riginoommen — CHANGES_REQUESTED** [review]

**Review by riginoommen — APPROVED** [review]

**Review by hybridx — APPROVED** [review]

**Pull request merged**